lib/repo/git_repository.rb

```diff
@@ -391,49 +470,65 @@ def get_users(permissions)
 
     def get_permissions(user_id)
 
-      #if @repos_admin # Are we admin?
-      # Adds a user with given permissions to the repository
+      if @repos_admin # Are we admin?
+        # Adds a user with given permissions to the repository
+        ga_repo = Gitolite::GitoliteAdmin.new(
+          Repository.conf[:REPOSITORY_PERMISSION_FILE])
 
-      ga_repo =
-          Gitolite::GitoliteAdmin.new(Repository.conf[:REPOSITORY_STORAGE])
-      repo = ga_repo.config.get_repo(self.get_repos.workdir.split('/').last)
+        # Sync the admin repo
+        ga_repo.update
+        repo = ga_repo.config.get_repo(self.get_repos.workdir.split('/').last)
 
-      # Gets permissions of a particular user
-      repo.permissions[0].each do |perm|
-        if(repo.permissions[0][perm[0]][""].include? user_id)
-          return self.class.__translate_perms_from_file(perm[0])
+        # Gets permissions of a particular user
+        repo.permissions[0].each do |perm|
+          if(repo.permissions[0][perm[0]][""].include? user_id)
```

Comment:
Use space after control keywords.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.